### PR TITLE
Fixed: Additional handling for obfuscated releases

### LINKED ISSUE #5680: Pull Sonarr commit 'Fixed: Additional handling for obfuscated releases'
Pull https://github.com/Sonarr/Sonarr/commit/471f0016b481647261f07084f019a155836062e0

```
git cherry-pick -ex 471f0016b481647261f07084f019a155836062e0
```

This commit has 2 conflict(s)

You can ask me to `close all`, `open pr here` or `open pr everywhere`

### PR DESCRIPTION
**Pull request by bakerboy448**
#### Database Migration
NO

#### Todos
- [X] Tests
- [] Translation Keys-N/A
- [] Wiki Updates-N/A

#### Issues Fixed or Closed by this PR

* Fixes #5680

### CODE REVIEW

**Comment by bakerboy448** [comment]
Adding additional entry for 39 char HASH (not in sonarr) 
https://discord.com/channels/264387956343570434/264388019585286144/798944310997418034

File: src/NzbDrone.Core.Test/ParserTests/HashedReleaseFixture.cs [hunk 1]
```diff
@@ -87,9 +87,22 @@ public class HashedReleaseFixture : CoreTest
...
+                "Movie Title",
+                Quality.HDTV720p,
+                null
+            },
+            new object[]
+            {
+                @"C:\Test\Movie Title 1964 Criterion Collection 1080p Blu-ray Remux AVC DTS-HD MS 1 0\zM0vnVU1SoV4uwTihfzTHkYmJPBemgN3MqXG1fY.mkv".AsOsAgnostic(),
+                "Movie Title",
+                Quality.Remux1080p,
+                null
```
Comment (Qstick) [inline src/NzbDrone.Core.Test/ParserTests/HashedReleaseFixture.cs:103]: These two cases pass just fine on develop currently without this change, given we don't get hung up on `s5E2` in the hash. So what does this actually do for us?

Comment (bakerboy448) [reply src/NzbDrone.Core.Test/ParserTests/HashedReleaseFixture.cs:103]: really just following precedent of the previous obfuscation tests and logic in there

Comment (Qstick) [reply src/NzbDrone.Core.Test/ParserTests/HashedReleaseFixture.cs:103]: Yea my point is why do we need this new hashed regex if its not doing anything for us.. at least based on the test cases added.

**Comment by bakerboy448** [comment]
closing per discussion; not needed as we are not impacted

**Pull request closed without merging**